src/firebase-resource-manager.ts

```diff
@@ -0,0 +1,154 @@
+  ORGANIZATION = "organization",
+  FOLDER = "folder",
+}
+
+export interface ParentResource {
+  id: string;
+  type: ParentResourceType;
+}
+
+export class FirebaseResourceManager {
+  private poller: LroPoller = new LroPoller();
+
+  async createFirebaseProject(
+    projectId: string,
+    projectDisplayName: string,
+    parentResource?: ParentResource
+  ): Promise<{ projectId: string }> {
+    await this.createCloudProject(projectId, projectDisplayName, parentResource);
+    const projectInfo = await this.addFirebaseToCloudProject(projectId);
+
+    logger.info("");
+    logger.info("🎉🎉🎉 Your Firebase project is ready! 🎉🎉🎉");
+    logger.info("");
+    logger.info("Project information:");
+    logger.info(`   - Project ID: ${clc.bold(projectInfo.projectId)}`);
+    logger.info(`   - Project Name: ${clc.bold(projectInfo.displayName)}`);
+    logger.info("");
+    logger.info("Firebase console is available at");
+    logger.info(`https://console.firebase.google.com/project/${clc.bold(projectId)}/overview`);
+    return { projectId };
+  }
+
+  /**
+   * Send an API request to create a new Google Cloud Platform project and poll the LRO to get the
+   * new project information.
+   * @return {Promise} this function returns a promise that resolves to the new cloud project
+   *     information
+   */
+  private async createCloudProject(
+    projectId: string,
+    projectDisplayName: string,
+    parentResource?: ParentResource
+  ): Promise<any> {
+    const spinner = new OraWrapper("Creating Google Cloud Platform project");
+    spinner.start();
+
+    try {
+      const response = await api.request("POST", "/v1/projects", {
+        auth: true,
+        origin: api.resourceManagerOrigin,
+        timeout: 15 * ONE_SECOND_MILLIS,
+        data: { projectId, name: projectDisplayName, parent: parentResource },
+      });
+      const projectInfo = await this.pollCloudProjectCreationOperation(
+        response.body.name /* LRO resource name */
+      );
+      spinner.succeed();
+      return projectInfo;
+    } catch (err) {
+      spinner.fail();
+      logger.debug(err.message);
+      throw new FirebaseError(
+        "Failed to create Google Cloud project. See firebase-debug.log for more info.",
+        { exit: 2, original: err }
+      );
+    }
+  }
+
+  /**
+   * @return {Promise} this function returns a promise that resolves to the new cloud project
+   *     information
+   */
+  private async pollCloudProjectCreationOperation(operationResourceName: string): Promise<any> {
+    const pollerOptions: LroPollerOptions = {
+      pollerName: "Project Creation Poller",
+      apiOrigin: api.resourceManagerOrigin,
+      apiVersion: "v1",
+      operationResourceName,
+    };
+    const { error, response } = await this.poller.poll(pollerOptions);
+    if (error) {
+      throw error;
+    }
+    return response;
+  }
+
+  /**
+   * Send an API request to add Firebase to the new Google Cloud Platform project and poll the LRO
```

Comment:
nit: Remove "new" before "Google Cloud Platform", since this API can be called with both new and old Cloud projects.

Reply:
Done